pkg/common/telemetry/config.go

```diff
@@ -40,3 +42,8 @@ type M3Config struct {
 	Env        string   `hcl:"env"`
 	UnusedKeys []string `hcl:",unusedKeys"`
 }
+
+type InMem struct {
+	Enabled    *bool    `hcl:"enabled"`
+	UnusedKeys []string `hcl:",unusedKeys"`
```

Comment:
UnusedKeys needs to be checked by both the agent and server CLI code like those in the other config blocks.

Reply:
darn, this isn't hcl magic? :)